### PR TITLE
D8 nid 1011

### PR DESCRIPTION
**Pull request by DuttonMa**
- Fixes to revision creation from draft

### CODE REVIEW

File: origins_workflow/origins_workflow.module [hunk 1]
```diff
@@ -336,7 +336,7 @@ function origins_workflow_preprocess_table(&$variables) {
           // Check the existing text in an attempt to detect any core fix that
           // might come along.
           $title_text = $title_markup->getUntranslatedString();
-          if (($title_text == 'Set as current revision') || ($title_text == 'Revert')) {
+          if ($title_text == 'Set as current revision') {
```
Comment (DuttonMa) [inline origins_workflow/origins_workflow.module:339]: Decided that there is no need to replace the core Drupal 'Revert' functionality as it is working correctly, it is only the 'Set as current revision' behaviour that we actually need to override.

File: origins_workflow/src/Controller/ModerationStateController.php [hunk 2]
```diff
@@ -102,9 +102,8 @@ private function transitionAllowed($entity, String $new_state) {
     // Check permissions of current user.
     $current_user = $this->currentUser();
     $transition_allowed = FALSE;
-    if (($current_state == 'draft') && ($new_state == 'published')) {
-      // Is this user allowed to use the 'quick publish' transition ?
-      if ($current_user->hasPermission('use editorial transition quick_publish')) {
+    if (($current_state == 'draft') && ($new_state == 'draft')) {
+      if ($current_user->hasPermission('use editorial transition create_new_draft')) {
         $transition_allowed = TRUE;
```
Comment (DuttonMa) [inline origins_workflow/src/Controller/ModerationStateController.php:107]: This permission check was missing and was causing the watchdog error that was mentioned in the ticket

File: origins_workflow/src/Controller/ModerationStateController.php [hunk 3]
```diff
@@ -113,6 +112,18 @@ private function transitionAllowed($entity, String $new_state) {
...
+      if ($current_user->hasPermission('use editorial transition quick_publish')) {
+        $transition_allowed = TRUE;
+      }
+    }
+    elseif (($current_state == 'published') && ($new_state == 'published')) {
+      // Is this user allowed to use the 'quick publish' transition ?
+      if ($current_user->hasPermission('use editorial transition quick_publish')) {
+        $transition_allowed = TRUE;
+      }
+    }
```
Comment (DuttonMa) [inline origins_workflow/src/Controller/ModerationStateController.php:126]: Several other state transitions that weren't previously covered have been added.

**Review by omahm — APPROVED** [review]
LG, nice work

**Pull request merged**